src/pages/ResultPage/ResultPage.tsx

```diff
@@ -0,0 +1,25 @@
+import { useLocation, useNavigate } from "react-router-dom";
+
+const ResultPage = () => {
+  const navigate = useNavigate();
+  const location = useLocation();
+
+  const isFailed = location.state?.isFailed;
+  const mistakesNumber = location.state?.mistakesNumber;
+
+  const onRetry = () => navigate("/");
+
+  return isFailed ? (
+    <div className="failed">
+      <p>Ooops... you failed, you did more than {mistakesNumber} mistake(s)</p>
+      <button onClick={onRetry}>pass the test again</button>
```

Comment:
It is more easier to use "&lt;Link to="/"&gt;" rather than additional function with button.

Reply:
fixed